### PR TITLE
travis: Add i586 linux and i686 musl

### LINKED ISSUE #35599: i686-musl + catch_unwind = fatal runtime error: failed to initiate panic, error 5
### STR

``` rust
use std::panic;

fn main() {
    let _ = panic::catch_unwind(|| panic!());
}
```

```
$ RUST_BACKTRACE=1 cargo run --target i686-unknown-linux-musl
    Finished debug [unoptimized + debuginfo] target(s) in 0.0 secs
     Running `target/i686-unknown-linux-musl/debug/foo`
thread 'main' panicked at 'explicit panic', src/main.rs:4
stack backtrace:
fatal runtime error: failed to initiate panic, error 5
error: Process didn't exit successfully: `target/i686-unknown-linux-musl/debug/foo` (signal: 6, SIGABRT: process abort signal)
```
### Meta

```
$ rustc -V
rustc 1.12.0-nightly (0ef24eed2 2016-08-10)

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04 LTS
Release:        16.04
Codename:       xenial
```

cc @alexcrichton

### LINKED ISSUE #39053: i686 musl cannot panic
e.g. for a simple program like:

```rust
fn main() {
   panic!("hello");
}
```

it runs as:

```
$ rustc foo.rs --target i686-unknown-linux-musl && ./foo
thread 'main' panicked at 'hello', foo.rs:2
note: Run with `RUST_BACKTRACE=1` for a backtrace.
fatal runtime error: failed to initiate panic, error 5
zsh: abort (core dumped)  ./foo
```

cc @japaric

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit expands the existing x86_64-musl entry in the Travis matrix to also
build/test i586-unknown-linux-gnu and i686-unknown-linux-musl.

cc #38531 
Closes #35599
Closes #39053

### CODE REVIEW

File: src/ci/docker/linux-tested-targets/build-musl.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+# Whoa what's this mysterious patch we're applying to libunwind! Why are we
+# swapping the values of ESP/EBP in libunwind?!
+#
+# Discovered in #35599 it turns out that the vanilla build of libunwind is not
+# suitable for unwinding 32-bit musl. After some investigation it ended up
+# looking like the register values for ESP/EBP were indeed incorrect (swapped)
+# in the source. Similar commits in libunwind (r280099 and r282589) have noticed
+# this for other platforms, and we just need to realize it for musl linux as
+# well.
```
Comment (brson) [inline src/ci/docker/linux-tested-targets/build-musl.sh]: holy shit

**Comment by alexcrichton** [comment]
@bors: retry

* network error

**Comment by alexcrichton** [comment]
@bors: r-

* i686 musl still has a broken unwinder

**Comment by alexcrichton** [comment]
With a tip from the #llvm IRC channel looks like all we needed was to enable frame pointers. I have done so and it fixes the failures I previously saw. The fix missed the most recent revision of the rollup, so I'll probably send a follow-up PR to enable testing for i686 musl.

**Pull request merged**